crates/ruff_linter/src/rules/airflow/rules/function_signature_change_in_3.rs

```diff
@@ -0,0 +1,141 @@
+use crate::checkers::ast::Checker;
+use crate::rules::airflow::helpers::Replacement;
+use crate::{Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{ViolationMetadata, derive_message_formats};
+use ruff_python_ast::{Arguments, Expr, ExprAttribute, ExprCall};
+use ruff_python_semantic::Modules;
+use ruff_python_semantic::analyze::typing;
+use ruff_text_size::Ranged;
+
+/// ## What it does
+/// Checks for Airflow function calls that will raise a runtime error in Airflow 3.0
+/// due to function signature changes, such as functions that changed to accept only
+/// keyword arguments, parameter reordering, or parameter type changes.
+///
+/// ## Why is this bad?
+/// Airflow 3.0 might introduce changes to function signatures. Code that
+/// worked in Airflow 2.x may raise a runtime error if not updated in Airflow
+/// 3.0.
+///
+/// ## Example
+/// ```python
+/// from airflow.lineage.hook import HookLineageCollector
+///
+/// collector = HookLineageCollector()
+/// # Passing positional arguments will raise a runtime error in Airflow 3.0
+/// collector.create_asset("s3://bucket/key")
+/// ```
+///
+/// Use instead:
+/// ```python
+/// from airflow.lineage.hook import HookLineageCollector
+///
+/// collector = HookLineageCollector()
+/// # Passing arguments as keyword arguments instead of positional arguments
+/// collector.create_asset(uri="s3://bucket/key")
+/// ```
+#[derive(ViolationMetadata)]
+#[violation_metadata(stable_since = "0.14.11")]
+pub(crate) struct Airflow3FunctionSignatureChange {
+    function_def: String,
+    replacement: Replacement,
+}
+
+impl Violation for Airflow3FunctionSignatureChange {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let Airflow3FunctionSignatureChange {
+            function_def,
+            replacement,
+        } = self;
+        match replacement {
```

Comment:
Do you thing we want a new enum or keep using `Replacement`? I feel like it's less likely to reuse things like `SourceModuleMoved`